backend/src/rss.rs

```diff
@@ -167,19 +184,13 @@ async fn video_items(
 
 
 
-/// This returns a track that:
+/// This returns a single track for use in the enclosure, that:
 /// a) is a `presentation` track.
-/// Defaults to any track meeting the b) criteria if there is no `presentation` track.
 /// b) has a resolution that is closest to full hd.
-fn preferred_tracks(tracks: Vec<EventTrack>) -> (Vec<EventTrack>, Option<EventTrack>) {
-    let target_resolution = tracks.first().map_or([1920, 1080], |first_track| {
-        let [x, y] = first_track.resolution.unwrap();
-        if x >= y {
-            [1920, 1080] // Landscape
-        } else {
-            [1080, 1920] // Portrait
-        }
-    });
+/// Defaults to any track meeting the b) criteria if there is no `presentation` track.
+/// It also returns a hashmap of all tracks grouped by their flavor.
+fn preferred_tracks(tracks: Vec<EventTrack>) -> (Option<EventTrack>, HashMap<String, Vec<EventTrack>>) {
```

Comment:
```suggestion
fn preferred_tracks(tracks: Vec<EventTrack>) -> (EventTrack, HashMap<String, Vec<EventTrack>>) {
```

Currently you `unwrap` in the calling function. The unwrap is actually ok as the DB ensures there is always at least one track per event. But then we can also do the unwrap in this function and don't need to return an `Option`. Specifically, `.unwrap()` after the `min_by_key` call below. Well, even better, use `.expect("event without tracks")` instead of `unwrap`: that immediately signals to code readers that this shouldn't happen.